### PR TITLE
Tighten conversation max recipients and use a helper to determine it

### PR DESCRIPTION
**Pull request by linc**
This addresses a potential spam vector where a bot could send hundreds of private messages if a site does not have an upper threshold set manually for the number of users a private message can be sent to.

* Refactor our max recipient checking into a helper method.
* Use a default max of FIVE for new users.
* Use a default max of FIFTY for verified users.
* Moderators may still send unlimited messages.

Future work must still be done to make our flood control tigther, but this is a good first step.

Backport to `release/2.3`.

### CODE REVIEW

**Comment by linc** [comment]
Paging @adrianspeyer to review the concept.

**Comment by adrianspeyer** [comment]
Makes a lot of sense. Especially the flood control, which seemed not to function for this. Solid!

**Comment by tburry** [comment]
I think we'll get into a bit of hot water if we can't override both of the default max numbers.

**Comment by linc** [comment]
You want the current global override to be per-default instead?

**Comment by tburry** [comment]
Yeah I think so. I don't want the case where people override the verified default to something smaller and that ends up increasing the non-verified default. Very strange behavior there.

**Comment by linc** [comment]
Added a second config for verified users to graduate to.

File: applications/conversations/controllers/class.messagescontroller.php [hunk 1]
```diff
@@ -60,12 +60,16 @@ public function add($Recipient = '', $Subject = '') {
         $this->permission('Conversations.Conversations.Add');
         $this->Form->setModel($this->ConversationModel);
 
-        // Set recipient limit
-        if (!checkPermission('Garden.Moderation.Manage') && c('Conversations.MaxRecipients')) {
-            $this->addDefinition('MaxRecipients', c('Conversations.MaxRecipients'));
-            $this->setData('MaxRecipients', c('Conversations.MaxRecipients'));
+        // Detect our recipient limit.
+        $maxRecipients = self::getMaxRecipients();
```
Comment (initvector) [inline applications/conversations/controllers/class.messagescontroller.php]: `ConversationModel::getMaxRecipients`

File: applications/conversations/models/class.conversationmodel.php [hunk 2]
```diff
@@ -296,6 +296,30 @@ public function getID($ConversationID, $datasetType = false, $options = []) {
     }
 
     /**
+     * Get how many recipients current user can send a message to.
+     *
+     * @return int|bool A maximum number of recipients or FALSE for unlimited.
+     */
+    public static function getMaxRecipients() {
+        // Moderators can add as many as they want.
+        if (checkPermission('Garden.Moderation.Manage')) {
```
Comment (initvector) [inline applications/conversations/models/class.conversationmodel.php]: What about using `Gdn_Session::checkRankedPermission` instead?

**Pull request merged**